presto-hive/src/main/java/io/prestosql/plugin/hive/orc/HdfsOrcDataSource.java

```diff
@@ -54,6 +56,15 @@ public void close()
         inputStream.close();
     }
 
+    @Override
+    public Slice readTail(int length)
+            throws IOException
+    {
+        //  Handle potentially imprecise file lengths by reading the footer
```

Comment:
Should probably record bytes read and timing like readInternal does.

Reply:
Yep.  Missed it in the parquet code also.